packages/p2p/stores/general-store.js

```diff
@@ -189,31 +196,42 @@ export default class GeneralStore {
     }
 
     @action.bound
-    setChatInfo(user_id, token) {
-        this.chat_info = {
-            app_id: isProduction() ? '1465991C-5D64-4C88-8BD9-B0D7A6455E69' : '4E259BA5-C383-4624-89A6-8365E06D9D39',
-            user_id: user_id,
-            token: token,
-        };
+    setChatInfo(p2p_advertiser_info_response) {
+        if (p2p_advertiser_info_response.error) {
+            this.ws_subscriptions.advertiser_subscription?.unsubscribe();
+        } else {
+            const { p2p_advertiser_info } = p2p_advertiser_info_response;
+
+            const getSendbirdServiceToken = () => {
+                requestWS({ service: 'sendbird', service_token: 1 }).then(response => {
+                    const { service_token } = response;
+
+                    runInAction(() => {
+                        this.chat_info = {
+                            app_id: getSocketURL().endsWith('binaryws.com')
+                                ? '1465991C-5D64-4C88-8BD9-B0D7A6455E69'
+                                : '4E259BA5-C383-4624-89A6-8365E06D9D39',
+                            user_id: p2p_advertiser_info.chat_user_id,
+                            token: service_token.sendbird.token,
+                        };
+                    });
 
-        if (!this.chat_info.token) {
-            requestWS({ service_token: 1, service: 'sendbird' }).then(response => {
-                this.chat_info.token = response.service_token.sendbird.token;
-            });
-        }
-    }
+                    // Refresh chat token ±1 hour before it expires (BE will refresh the token
+                    // when we request within 2 hours of the token expiring)
+                    const expiry_moment = epochToMoment(service_token.sendbird.expiry_time);
+                    const delay_ms = expiry_moment.diff(
+                        this.props.server_time
+                            .get()
+                            .clone()
+                            .subtract(1, 'hour')
+                    );
+
+                    this.timeouts.push(setTimeout(getSendbirdServiceToken, delay_ms));
```

Comment:
This will push to `timeouts` every time the request is sent, do we want to handle what happens to the previous interval here? We're currently only handling this in `onUnmount`.

Reply:
Sorry about the confusion between timeouts and intervals (due to the clearInterval that was previously there). I've simplified this line and it should be more understandable now. 🙏